### PR TITLE
Add is.domElement()

### PR DESCRIPTION
**Pull request by melvin0008**
Check if the value passed is a dom Element or not.

Reference: https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

### CODE REVIEW

**Comment by melvin0008** [comment]
Reference: https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

File: readme.md [hunk 1]
```diff
@@ -145,6 +145,10 @@ Check if `value` (number) is in the range of `0` to `upperBound`.
 is.inRange(3, 10);
 ```
 
+##### .domElement(value)
+
+Returns `true` if `value` is a domElement.
```
Comment (kevva) [inline readme.md]: > Returns `true` if `value` is a DOM element.

File: readme.md [hunk 2]
```diff
@@ -145,6 +145,10 @@ Check if `value` (number) is in the range of `0` to `upperBound`.
 is.inRange(3, 10);
 ```
 
+##### .domElement(value)
```
Comment (kevva) [inline readme.md]: Maybe `.domNode`? Just a question.

Comment (melvin0008) [reply readme.md]: There are 7 types of nodes supported currently .
Reference : https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Element is one of them. So I feel .domElement() is more appropriate for this pull request

However I create a different function .domNode() which will check if value is a node?

Comment (kevva) [reply readme.md]: > Element is one of them. So I feel .domElement() is more appropriate for this pull request

Makes sense.

> However I can create a different function .domNode() which will check if value is a node or not?

I don't think its use case is that common, but I'll let @sindresorhus decide if he wants it.

File: index.js [hunk 3]
```diff
@@ -161,4 +161,6 @@ is.inRange = (x, range) => {
 	throw new TypeError(`Invalid range: ${util.inspect(range)}`);
 };
 
+is.domElement = x => is.object(x) && x.nodeType === 1;
```
Comment (sindresorhus) [inline index.js]: I think this is a bit too loose of a check.

Not sure what else we should check, but see https://www.google.co.th/search?q=is+dom+element+javascript+check&oq=is+dom+element+javascript+check&aqs=chrome..69i57j0l2j69i64l3.4818j0j1&sourceid=chrome&ie=UTF-8 and https://github.com/lodash/lodash/blob/master/isElement.js for some ideas.

File: readme.md (hunk 2 above)
Comment (sindresorhus) [reply readme.md]: > I don't think its use case is that common, but I'll let @sindresorhus decide if he wants it.

Nah, only interested in `domElement`.

File: index.js (hunk 3 above)
Comment (melvin0008) [reply index.js]: Yes, I do agree now. Any object with a nodeType would be selected as a domElement. I will change this behavior and update the pull request.

**Comment by melvin0008** [comment]
I added additional checks to make sure we do not accept plain objects. 

However if a object looks like this 

```js
class Foo {
    get nodeType() {
	return 1;
  }
  get nodeName() {
	return 'div';
  }
}

is.domElement(new Foo())
```

it will return true. It also returns true in lodash.

**Comment by sindresorhus** [comment]
Maybe we should check if a few more properties exist, just be sure. For example, these:

- `.className`
- `.style`
- `.ownerDocument`
- `.innerHTML`
- `.setAttribute`

File: test.js [hunk 4]
```diff
@@ -358,3 +359,18 @@ test('is.inRange', t => {
...
+	t.true(m.domElement(document.createElement('div')));
+	t.false(m.domElement('hello world'));
+	t.false(m.domElement([]));
+	t.false(m.domElement(new Map()));
+	t.false(m.domElement(null));
+	t.false(m.domElement(undefined));
+	t.false(m.domElement(0));
+	t.false(m.domElement(NaN));
+	t.false(m.domElement(Infinity));
+	t.false(m.domElement({nodeType: 1, nodeName: 'div'}));
```
Comment (sindresorhus) [inline test.js]: Can you use the `testType` helper?

Comment (melvin0008) [reply test.js]: Yes sure. I'll update the review

File: index.js [hunk 5]
```diff
@@ -161,4 +161,10 @@ is.inRange = (x, range) => {
 	throw new TypeError(`Invalid range: ${util.inspect(range)}`);
 };
 
+is.domElement = x => {
+	const propsToCheck = ['innerHTML', 'ownerDocument', 'style', 'attributes', 'nodeValue'];
+	return is.object(x) && x.nodeType === 1 && is.string(x.nodeName) &&
```
Comment (gioragutt) [inline index.js]: what does 1 stand for here? please convert `"magic numbers"` to consts.

File: test.js [hunk 6]
```diff
@@ -89,7 +92,8 @@ const types = new Map([
 		Object.create(null),
 		new Object() // eslint-disable-line no-new-object
 	]],
-	['integer', 6]
+	['integer', 6],
+	['domElement', document.createElement('div')]
```
Comment (gioragutt) [inline test.js]: What about more than a div?

you can do `['div', 'input', 'span', 'img', 'canvas', 'h1', 'h2', 'h3' ...].map(e => document.createElement(e))`

Comment (sindresorhus) [reply test.js]: Agreed, but we don't need the different `h` tags.

**Comment by gioragutt** [comment]
I had just given some examples as a guideline, there are plenty more
examples.

בתאריך 5 באוק׳ 2017 5:02 אחה״צ,‏ "Sindre Sorhus" <notifications@github.com>
כתב:

File: test.js (hunk 6 above)
Comment (sindresorhus) [reply test.js]: Let's go with: `div`, `input`, `span`, `img`, `canvas`, `script`.

Comment (gioragutt) [reply test.js]: Also, what markup object is not considered a domElement? So that we can check for more cases

**Comment by melvin0008** [comment]
@gioragutt Thanks for the review. Learnt few things there

**Comment by gioragutt** [comment]
@melvin0008 glad to help dude, thanks for the PR!

@sindresorhus any reason not to merge this?

**Comment by sindresorhus** [comment]
@gioragutt I reviewed it and then got distracted and forgot about it.

@melvin0008 Nice work! :)

**Comment by melvin0008** [comment]
Thank You:)

**Pull request merged**